how-it-works/3_chainevolution/01_scalability.subpage.md

```diff
@@ -0,0 +1,27 @@
+---
+title: Infinite scalability
+abstract: 
+shareImage: /img/how-it-works/resumption.600.jpg
+slug: scalability
+---
+
+# Infinite scalability
+
+Ever wondered the meaning behind DFINITY? It’s Decentralized + Infinity. It’s named that way because the Internet Computer is designed to scale infinitely. It means that the Internet Computer have host unlimited number of canisters (smart contracts), store unlimited amount of memory, process unlimited amount of transactions per second. In simple words, Internet Computer is designed to host even large scale social media platforms in a fully decentralized way. 
+
+There are two types of widely-used approaches to improve the scalability of a system. (1) Vertical Scaling, and (2) Horizontal Scaling. Vertical scaling means adding more CPU, RAM and disk to a single computer. Horizontal scaling means adding more computers to the system. There is a limit to vertical scaling. But with horizontal scaling, one can achieve unlimited scalability. Internet Computer is one of the first blockchains to successfully use horizontal scaling. 
+
+The nodes in the Internet Computer are divided into subnets, each containing a few dozen nodes. The set of nodes in a subnet together maintain one blockchain. Each subnet can host a few thousand canisters and process messages received by those canisters. Each subnet has a limited capacity in terms of the number of canisters (a few thousand), amount of storage (a few TB), and bandwidth (a few hundred transactions per second). But as more subnets are added to the Internet Computer, its overall capacity increases proportionately. There is no limit on the number of subnets that can be added, resulting in unlimited scalability. 
+
+Creating a new subnet has two steps.  1) Adding new nodes to the Internet Computer,  and  2) Creating a subnet with the available nodes
+Anyone can purchase the node hardware and add it to the Internet Computer by following the [node provider onboarding process](https://wiki.internetcomputer.org/wiki/Node_Provider_Onboarding). 
+
+We now describe how to create a new subnet with the available nodes. The Internet Computer has a decentralized governance system called Network Nervous System (NNS). Essentially, NNS consists of a group of canisters that manage the Internet Computer. In the NNS, there is a component called “registry”, which stores all the configuration of the Internet Computer. The registry has a record for each subnet which includes a protocol version, list of nodes in the subnet, cryptographic key material to be used by the subnet, a Catch Up Package (CUP), etc. 
```

Comment:
>he registry has a record for each subnet which includes [...] cryptographic key material to be used by the subnet

Can this be misconstrued as the NNS holding private keys of another subnet?

Reply:
That may need some clarification indeed. I'll do that.